packages/wrangler/src/pages/upload.tsx

```diff
@@ -368,6 +369,13 @@ export const upload = async (
 	);
 };
 
+function isJwtExpiredError(error: unknown): boolean {
+	return (
+		error instanceof ParseError &&
+		error?.notes.some((m) => m.text === "Expired JWT")
```

Comment:
This is probably good enough for now, but I'd much rather we use an error code in case this messaging ever changes

Reply:
I agree, to check for a specific string in this type of scenario is not a bulletproof solution by any means, and normally I would avoid it, but after spending a bit of time debugging the chain of functions and errors, none of the errors or responses return any kind of error code, which is why the existing check does not refetch the JWT token. 
I could be missing something though, definitely open to some direction or guidance to get this in a good spot and be able to merge.

Reply:
> none of the errors or responses return any kind of error code

That's no good :) Looking at the API code, we're _supposed to be returning_ a response body that looks like this:

```tsx
{
  success: false,
  errors: [
    {
      code: 8000013,
      message: 'Authorization failed',
    },
  ],
}
```

But I'm guessing that's getting lost in translation somewhere.

Another approach we could consider for fixing this particular problem is to parse the JWT and just check locally if it's expired or not

Reply:
Ah, that would make perfect sense. 

Cool, can do. I should be able to make some headway on this towards the end of the week.